drivers/sensor/lsm6dso/lsm6dso.c

```diff
@@ -365,6 +374,12 @@ static int lsm6dso_sample_fetch_temp(const struct device *dev)
 #if defined(CONFIG_LSM6DSO_SENSORHUB)
 static int lsm6dso_sample_fetch_shub(const struct device *dev)
 {
+	struct lsm6dso_data *data = dev->data;
+
+	if (!data->shub_inited) {
+		return 0;
```

Comment:
I do not return an error intentionally here. But I slightly changed my mind now. I think that the driver should return error here, because an app should not call fetch for shub for a given instance which is not a sensorhub (or at least should be reported accordingly that it isn't). So, I'll think more about it.

Reply:
@MaureenHelm 
I removed this check here. INstead I call the routine only if shub_inited is true. Is basically the same as the above code, but it is maybe more clear (i.e. entering shub_data_fetch() only if shub is inited, instead of exiting w/o error if shub is not inited).
What do you  think? Is it convincing?